cobra/test/test_flux_analysis.py

```diff
@@ -391,73 +391,71 @@ def test_single_gene_deletion_linear_room_benchmark(self, model,
+        # Expect the ROOM solution to have smaller flux changes in
         # reactions compared to a normal FBA.
-        assert abs(active - room_active) <= abs(active - knock_active)
-        # Expect the FBA-based reference to have more active reactions.
-        assert room_ref_active >= room_active
+        assert flux_change_room <= flux_change
+        # Expect the FBA-based reference to have less change in
+        # flux distribution.
+        assert flux_change_room_ref >= flux_room_ref
 
     @pytest.mark.skip("Unpredictable solutions.")
     @pytest.mark.parametrize("solver", optlang_solvers)
-    def test_linear_room_sanity(self, solver, threshold=0.0):
-        model = construct_papin_2003_model()
+    def test_linear_room_sanity(self, solver):
         model.solver = solver
         sol = model.optimize()
-        active = (sol.fluxes.abs() > threshold).sum()
         with model:
-            model.reactions.v3.knock_out()
+            model.reactions.PYK.knock_out()
             knock_sol = model.optimize()
-            knock_active = (knock_sol.fluxes.abs() > threshold).sum()
 
         with model:
             # Internally uses pFBA as reference solution.
             add_room(model, linear=True)
-            model.reactions.v3.knock_out()
+            model.reactions.PYK.knock_out()
             room_sol = model.optimize()
-            room_active = (room_sol.fluxes.abs() > threshold).sum()
 
         with model:
             # Use FBA as reference solution.
             add_room(model, solution=sol, linear=True)
-            model.reactions.v3.knock_out()
+            model.reactions.PYK.knock_out()
             room_ref_sol = model.optimize()
-            room_ref_active = (room_ref_sol.fluxes.abs() > threshold).sum()
 
-        # Expect the ROOM solution to have a smaller number of active
+        flux_change = (sol.fluxes - knock_sol.fluxes).abs().sum()
+        flux_change_room = (sol.fluxes - room_sol.fluxes).abs().sum()
+        flux_change_room_ref = (sol.fluxes - room_sol_ref.fluxes).abs().sum()
+        # Expect the ROOM solution to have smaller flux changes in
         # reactions compared to a normal FBA.
-        assert abs(active - room_active) <= abs(active - knock_active)
-        # Expect the FBA-based reference to have more active reactions.
-        assert room_ref_active >= room_active
+        assert flux_change_room <= flux_change
+        # Expect the FBA-based reference to have less change in
+        # flux distribution.
+        assert flux_change_room_ref >= flux_room_ref
 
     @pytest.mark.parametrize("solver", optlang_solvers)
     def test_single_reaction_deletion_room(self, solver):
-        model = construct_papin_2003_model()
+        model = construct_room_model()
         model.solver = solver
-        sol = construct_papin_2003_solution()
+        sol = construct_room_solution()
         expected = Series({'v1': 10.0, 'v2': 5.0, 'v3': 0.0, 'v4': 5.0,
```

Comment:
Aren't the expected fluxes exactly those in the constructed solution? (Same in line 476.)